spec/javascripts/health-care/components/full_name.spec.jsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import ReactTestUtils from 'react-addons-test-utils';
+import FullName from '../../../../_health-care/_js/_components/full_name';
+import _ from 'lodash';
+
+describe('<FullName>', () => {
+  let component = null;
+
+  beforeEach(() => {
+    component = ReactTestUtils.renderIntoDocument(
+      <FullName name={{first:"William", last:"Shakespeare"}} onUserInput={(update) => {}}/>
+    );
+    assert.ok(component, "Cannot even render component");
+  });
+
+  it('has sane looking features', () => {
+    let inputs = ReactTestUtils.scryRenderedDOMComponentsWithTag(component, 'input');
+    expect(inputs).to.have.length(3);
+
+    let selects = ReactTestUtils.scryRenderedDOMComponentsWithTag(component, 'select');
+    expect(selects).to.have.length(1);
+  });
+
+  it('sets and removes error css on blank first name', () => {
+    // Initial state should be valid.
+    expect(component.state.hasError).to.be.false;
+    expect(ReactTestUtils.scryRenderedDOMComponentsWithClass(
+        component, 'usa-input-error')).to.have.length(0);
+
+    // Empty first name renders error.
+    component.refs.first.value = '';
+    ReactTestUtils.Simulate.change(component.refs.first);
+    expect(component.state.hasError).to.be.true;
+    expect(ReactTestUtils.scryRenderedDOMComponentsWithClass(
+        component, 'usa-input-error')).to.have.length(1);
+
+    // Present first name removes error.
+    component.refs.first.value = 'a';
+    ReactTestUtils.Simulate.change(component.refs.first);
+    expect(component.state.hasError).to.be.false;
+    expect(ReactTestUtils.scryRenderedDOMComponentsWithClass(
+        component, 'usa-input-error')).to.have.length(0);
+  });
+
+  it('validate first and last name are present', () => {
+    // Present first name and blank last name should render error.
+    component.refs.first.value = 'a';
```

Comment:
One thing to be careful of...I think you have to actually set every component before calling the onChange simulation otherwise it ends up with the default value you set.  You might want to throw some console.log() statements in the onChange handler to inspect what's going on to be certain.